victron_ble/devices/solar_charger.py

```diff
@@ -0,0 +1,90 @@
+from enum import Enum
+from typing import Any, Dict
+
+from construct import Int16sl, Int16ul, Struct
+
+from victron_ble.devices.base import Device, DeviceData
+
+
+class ChargeState(Enum):
+    OFF = 0
+    BULK = 3
+    ABSORPTION = 4
+    FLOAT = 5
```

Comment:
There is a complete table of these in the VE.Direct documentation, FWIW:

```
class ChargerState(Enum):
    OFF = 0
    LOW_POWER = 1  # Inverter-only
    FAULT = 2
    BULK = 3
    ABSORPTION = 4
    FLOAT = 5
    STORAGE = 6  # Charger-only
    EQUALIZE = 7
    INVERTING = 8  # Inverter-only
    POWER_SUPPLY = 11  # AC Charger-only
    STARTING_UP = 245
    REPEATED_ABSORPTION = 246  # AC Charger-only
    RECONDITION = 247
    BATTERY_SAFE = 248  # AC Charger-only
    EXTERNAL_CONTROL = 252
```

Reply:
The Solar Charger only has "off", "bulk"; "absorption" and "float".  No other modes are used.

Reply:
Ours has a manual equalization feature, so it's possible that  those would be seen

Reply:
I think it's worth just duplicating the enum from the docs since we'll presumably end up adding most of these as we start supporting other devices and I think we can share this enum across all the implementations

Reply:
Let's rename this to something more generic like OperationMode to match this table since these values aren't only for chargers:

<img width="674" alt="image" src="https://user-images.githubusercontent.com/484570/210013980-8ba8e228-0084-4243-8e33-f3f59031f7d4.png">


```
class OperationMode(Enum):
    OFF = 0
    LOW_POWER = 1  # AC Inverter-only
    FAULT = 2
    BULK = 3
    ABSORPTION = 4
    FLOAT = 5
    STORAGE = 6  # AC Charger-only
    EQUALIZE = 7
    INVERTING = 8  # Inverter-only
    POWER_SUPPLY = 11  # AC Charger-only
    STARTING_UP = 245
    REPEATED_ABSORPTION = 246  # AC Charger-only
    RECONDITION = 247
    BATTERY_SAFE = 248  # AC Charger-only
    EXTERNAL_CONTROL = 252
```

Reply:
> Let's rename this to something more generic like OperationMode to match this table since these values aren't only for chargers:
> 
> <img alt="image" width="674" src="https://user-images.githubusercontent.com/484570/210013980-8ba8e228-0084-4243-8e33-f3f59031f7d4.png">
> 
> ```
> class OperationMode(Enum):
>     OFF = 0
>     LOW_POWER = 1  # AC Inverter-only
>     FAULT = 2
>     BULK = 3
>     ABSORPTION = 4
>     FLOAT = 5
>     STORAGE = 6  # AC Charger-only
>     EQUALIZE = 7
>     INVERTING = 8  # Inverter-only
>     POWER_SUPPLY = 11  # AC Charger-only
>     STARTING_UP = 245
>     REPEATED_ABSORPTION = 246  # AC Charger-only
>     RECONDITION = 247
>     BATTERY_SAFE = 248  # AC Charger-only
>     EXTERNAL_CONTROL = 252
> ```

I think this should be defined in a more global way as this will be used by other devices in the future, so I don't want to squeeze this into solar_charger.py

Reply:
Maybe we can move this into base.py for now?